sample-app/src/hooks/useSynchronizedSearch.tsx

```diff
@@ -0,0 +1,24 @@
+import { useRef, useState } from "react";
+
+export function useSynchronizedSearch<T>(
```

Comment:
maybe `useSyncronizedRequest`? nothing here is specific to searching

Reply:
Updated search to request and simplified some naming